File: ext/src/main/java/com/mrboomdev/awery/ext/data/Video.java
package com.mrboomdev.awery.ext.data;

import java.io.Serial;
import java.io.Serializable;

public class Video implements Serializable {
	@Serial
	private static final long serialVersionUID = 1;
	private final String title, url, extra;

	public Video(String title, String url, String extra) {
		this.title = title;
		this.url = url;
		this.extra = extra;
	}

	public Video(String title, String url) {
		this(title, url, null);
	}

	public Video(String url) {
		this(url, url);
	}

	public String getExtra() {
		return extra;
	}

	public String getTitle() {
		return title;
	}

	public String getUrl() {
		return url;
	}
}

File: extension/sdk/src/commonMain/kotlin/com/mrboomdev/awery/extension/sdk/Video.kt
package com.mrboomdev.awery.extension.sdk

import kotlinx.serialization.Serializable

@Serializable
data class Video(
    val url: String,
    val title: String? = null,
    val videoTracks: List<Track> = emptyList(),
    val subtitleTracks: List<Track> = emptyList(),
    val audioTracks: List<Track> = emptyList(),
    val headers: Map<String, String> = emptyMap()
) {
    @Serializable
    data class Track(
        val url: String,
        val title: String? = null,
        val locale: String? = null
    )
}